sql-plugin/src/main/scala/com/nvidia/spark/rapids/aggregate.scala

```diff
@@ -396,11 +400,12 @@ class GpuHashAggregateIterator(
    */
   private def concatenateAndMerge(
       batches: mutable.ArrayBuffer[LazySpillableColumnarBatch]): LazySpillableColumnarBatch = {
-    withResource(batches) { _ =>
-      withResource(concatenateBatches(batches)) { concatBatch =>
-        withResource(computeAggregate(concatBatch, concatAndMergeHelper)) { mergedBatch =>
-          LazySpillableColumnarBatch(mergedBatch, metrics.spillCallback, "agg merged batch")
-        }
+    val concatBatch = withResource(batches) { _ =>
+      concatenateBatches(batches)
+    }
+    closeOnExcept(concatBatch) { _ =>
```

Comment:
Same double-close issue here.  computeAggregateAndClose is responsible for closing, so this should not also try to close in any scenario.